packages/phone-number-privacy/signer/test/integration/domain.test.ts

```diff
@@ -498,10 +513,35 @@ describe('domainService', () => {
           now: res2.body.status.now,
         },
       })
+      const eval2 = thresholdPoprfClient.unblindPartialResponse(
+        Buffer.from(res2.body.signature, 'base64')
+      )
+      expect(eval2).toEqual(eval1)
+    })
+
+    it('Should respond with 200 if nonce > domainState', async () => {
+      const [req, thresholdPoprfClient] = await signatureRequest(undefined, 1)
```

Comment:
Nit: consider submitting this request with nonce == 2 to make it extra clear and also so that we're testing whether counter just gets incremented vs. set all the way up to the nonce value. (i.e. will counter be returned as 1 or 2 if we submit with nonce 2)

Reply:
This is a somewhat subtle point that Victor and I discussed. The only reason the client should generate a request with a nonce higher than counter+1 is if it needs to in order to reach a threshold of signers (i.e. if the signers have different views of what counter the client is on). In this case, we want to preserve the user's quota on each signer in case the discrepancy in signer states changes direction again. A replay attack here could only deplete the user's quota up to the point that setting counter == nonce would anyway, and the replayed requests don't pose any additional security threat beyond depleting the user's quota unless a threshold of signers can be reached. It's a pretty subtle edge case and in general either behavior would be fine here. This is the expected behavior we landed on though IIRC. Let me know if you have any thoughts on this!

Reply:
Sounds good, thanks for the context!